### PR TITLE
Fix notices for empty database result sets

### PR DESCRIPTION
**Pull request by ausi**
| Q                | A
| -----------------| ---
| Fixed issues     | Fixes #1739 

Same as #1741 for the 4.9 branch

### CODE REVIEW

File: core-bundle/src/Resources/contao/library/Contao/Database/Result.php [hunk 1]
```diff
@@ -392,10 +392,15 @@ public function row($blnEnumerated=false)
...
+			if (!isset($this->resultSet[$this->intIndex]))
+			{
+				return array();
+			}
+
 			return $blnEnumerated ? array_values($this->resultSet[$this->intIndex]) : $this->resultSet[$this->intIndex];
 		}
 
-		$row = array_merge($this->resultSet[$this->intIndex], $this->arrModified);
+		$row = array_merge(isset($this->resultSet[$this->intIndex]) ? $this->resultSet[$this->intIndex] : array(), $this->arrModified);
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/library/Contao/Database/Result.php:403]:
```suggestion
		$row = $this->arrModified;

		if (isset($this->resultSet[$this->intIndex]))
		{
			$row = array_merge($this->resultSet[$this->intIndex], $this->arrModified);
		}
```

This way we would not call array_merge() at all if `$this->resultSet[$this->intIndex]` is not set. Not sure if that makes a difference though.

Comment (ausi) [reply core-bundle/src/Resources/contao/library/Contao/Database/Result.php:403]: I think in 99% of the cases `$this->arrModified` is empty and `$this->resultSet[$this->intIndex]` is set so I would prefer the current way which does not copy the `$this->arrModified` array.

**Review by leofeyer — DISMISSED** [review]

**Comment by leofeyer** [comment]
There is a PHPStan error though that needs to be fixed.

File: core-bundle/tests/Contao/Database/ResultTest.php [hunk 2]
```diff
@@ -25,16 +26,34 @@ public function testEmptyResult(): void
 
         /** @var Result|object $result */
```
Comment (leofeyer) [inline core-bundle/tests/Contao/Database/ResultTest.php:27]: This type-hint is causing the PHPStan error I believe.

Comment (ausi) [reply core-bundle/tests/Contao/Database/ResultTest.php:27]: fixed in 0fd67ddfb98ce85785aa11e79e3163c8e0fb9d68

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @ausi.

**Pull request merged**